pytest_tests/testsuites/conftest.py

```diff
@@ -134,35 +135,47 @@ def check_binary_versions(request, hosting: Hosting, client_shell: Shell):
 
 @pytest.fixture(scope="session")
 @allure.title("Prepare tmp directory")
-def temp_directory():
+def temp_directory() -> str:
     with allure.step("Prepare tmp directory"):
         full_path = os.path.join(os.getcwd(), ASSETS_DIR)
-        shutil.rmtree(full_path, ignore_errors=True)
-        os.mkdir(full_path)
+        create_dir(full_path)
 
     yield full_path
 
     with allure.step("Remove tmp directory"):
-        shutil.rmtree(full_path)
+        remove_dir(full_path)
 
 
 @pytest.fixture(scope="session", autouse=True)
-@allure.title("Collect logs")
-def collect_logs(temp_directory, hosting: Hosting):
+@allure.title("Collect full logs")
+def collect_full_tests_logs(temp_directory, hosting: Hosting):
+    test_name = 'full_logs'
     start_time = datetime.utcnow()
     yield
     end_time = datetime.utcnow()
-
-    # Dump logs to temp directory (because they might be too large to keep in RAM)
     logs_dir = os.path.join(temp_directory, "logs")
-    dump_logs(hosting, logs_dir, start_time, end_time)
-    attach_logs(logs_dir)
+    store_logs(hosting, logs_dir, test_name, start_time, end_time)
     check_logs(logs_dir)
 
 
+@pytest.fixture(scope="function", autouse=True)
+@allure.title("Collect test logs")
+def collect_test_logs(request, temp_directory, hosting: Hosting):
+    test_name = request.node.nodeid.replace(":", "_").replace("[", "_").replace("]", "_").replace("/", "_")
```

Comment:
`translate()`?

Reply:
I mean you can do `translate()` and not `replace().replace().replace().replace()`, it'd be more efficient and readable at the same time.

Reply:
> Linux does not allow such characters in file names

BTW, it's not true, all of these symbols are perfectly fine. Inconvenient in some cases, true, but valid:
```
$ touch 'test_container.py::TestContainer::test_container_creation[No name]'
$ ls -l 'test_container.py::TestContainer::test_container_creation[No name]'
-rw-r--r-- 1 rik users 0 июн 14 12:57 test_container.py::TestContainer::test_container_creation[No name]
```

Reply:
> > Linux does not allow such characters in file names
> 
> BTW, it's not true, all of these symbols are perfectly fine. Inconvenient in some cases, true, but valid:
> 
> ```
> $ touch 'test_container.py::TestContainer::test_container_creation[No name]'
> $ ls -l 'test_container.py::TestContainer::test_container_creation[No name]'
> -rw-r--r-- 1 rik users 0 июн 14 12:57 test_container.py::TestContainer::test_container_creation[No name]
> ```

Yes, but it is parsed inside tests without quotes, and the path is part of the return value.

I changed to translate(), if it critical, I can change the name to hash.
Anyway, in allure report attached files are named differently, it's important only for local runs and parsing test crashes.